tdrs-backend/tdpservice/data_files/models.py

```diff
@@ -1,13 +1,50 @@
 """Define data file models."""
+import logging
 import os
+from hashlib import sha256
+from io import StringIO
+from typing import Union
 
+from django.contrib.admin.models import ADDITION, ContentType, LogEntry
+from django.core.files.base import File
 from django.db import models
 from django.db.models import Max
 
 from tdpservice.backends import DataFilesS3Storage
 from tdpservice.stts.models import STT
 from tdpservice.users.models import User
 
+logger = logging.getLogger(__name__)
+
+def get_file_shasum(file: Union[File, StringIO]) -> str:
```

Comment:
I moved this here to avoid a circular dependency issue. The security model requires the data_files/models.py module